src/helpers/convertToIcns.ts

```diff
@@ -0,0 +1,49 @@
+import * as path from 'path';
+import * as tmp from 'tmp';
+
+import * as shell from 'shelljs';
+
+import { isOSX } from './helpers';
+
+tmp.setGracefulCleanup();
+
+const PNG_TO_ICNS_BIN_PATH = path.join(__dirname, '../..', 'bin/convertToIcns');
+
+/**
+ * Converts the png to a temp directory which will be cleaned up on process exit
+ */
+export function convertToIcns(
+  pngSource: string,
+  callback: (error: any, iconsDestination: string) => void,
+): void {
+  const tmpIconDirPath = tmp.dirSync({ unsafeCleanup: true }).name;
+  if (!isOSX()) {
+    callback('OSX is required to convert a .png icon to .icns', pngSource);
+    return;
+  }
+
+  const iconsDestination = `${tmpIconDirPath}/icon.icns`;
+  shell.exec(
+    `"${PNG_TO_ICNS_BIN_PATH}" "${pngSource}" "${iconsDestination}"`,
+    { silent: true },
+    (exitCode, stdOut, stdError) => {
+      if (stdOut.includes('icon.iconset:error') || exitCode) {
+        if (exitCode) {
+          callback(
+            {
+              stdOut,
+              stdError,
+            },
+            pngSource,
+          );
+          return;
+        }
+
+        callback(stdOut, pngSource);
+        return;
+      }
+
+      callback(null, iconsDestination);
+    },
+  );
+}
```

Comment:
> I can't tell if this file is still doing anything... It looks like it should have been removed during one of my earlier refactors, and it might already be replaced by iconShellHelpers.ts.

@jiahaog nice catch! Yes it's now properly handled by iconShellHelpers.ts . Removing.